File: List Permutations(Java_C)/Pa2(C)/ListTest.c

/**********************************************************************************
* Eloy Salinas
* eloys@ucsc.edu | 1243517
* Pa2
* ListTest.c
* A test client for List ADT
***********************************************************************************/

#include<stdio.h>
#include"List.h"

int main(int argc, char* argv[])
{
   int i;
   ListRef A = newList();
   ListRef B = newList();

   for(i=1; i<=10; i++)
   {
      insertBack(A, i);
      insertBack(B, 11-i);
   }
   printListStd(A);
   printListStd(B);
   printf("\n");

   for(i=1; i<=6; i++)
   {
      insertBack(A, getFront(B));
      deleteFront(B);
   }
   printListStd(A);
   printListStd(B);
   printf("\n");

   printf("%s\n", equals(A,B)?"true":"false");
   printf("%s\n", equals(A,A)?"true":"false");
   printf("\n");
   printf("offEnd(A) %d", offEnd(A));
   printf("\n");
   printf("getIndex(A) %d", getIndex(A));
   printf("\n");
   printf("getBack(A) %d", getBack(A));
   printf("\n");
   /*printf("getCurrent(A) %d", getCurrent(A));*/
   printf("\n");

   ListRef C = newList();
   insertBack(C, 4);
   insertBack(C, 3);
   insertBack(C, 2);
   insertBack(C, 1);

   printListStd(C);
   printf("%s\n", equals(B,C)?"true":"false");
   printf("\n");
   makeEmpty(C);
   printf("isEmpty(C) ");
   printf("%s\n", isEmpty(C)?"true":"false");

   moveTo(B, 1);
   printf("getCurrent(B) %d", getCurrent(B));
   printf("\n");
   printf("getLength(B) %d", getLength(B));
   printf("\n");
   printf("getIndex(B) %d", getIndex(B));
   movePrev(B);
   printf("\n");
   printf("getCurrent(B) %d", getCurrent(B));
   moveNext(B);
   moveNext(B);
   printf("\n");
   printf("getCurrent(B) %d", getCurrent(B));
   printf("\n");
   printf("getIndex(B) %d", getIndex(B));
   moveNext(B);
   printf("\n");
   printf("getCurrent(B) %d", getCurrent(B));
   printf("\n");
   printf("getIndex(B) %d", getIndex(B));
   insertFront(B, 9);
   printf("\n");
   printListStd(B);
   printf("\n");
   printf("getCurrent(B) %d", getCurrent(B));
   printf("\n");
   printf("getIndex(B) %d", getIndex(B));
   insertBeforeCurrent(B, 8);
   printf("\n");
   printListStd(B);
   printf("\n");
   printf("getCurrent(B) %d", getCurrent(B));
   printf("\n");
   printf("getIndex(B) %d", getIndex(B));
   moveTo(B, 3);
   insertAfterCurrent(B, 5);
   printf("\n");
   printListStd(B);
   printf("\n");
   printf("getCurrent(B) %d", getCurrent(B));
   printf("\n");
   printf("getIndex(B) %d", getIndex(B));
   printf("\n");
   printf("getLength(B) %d", getLength(B));
   deleteFront(B);
   deleteBack(B);
   printf("\n");
   printListStd(B);
   printf("\n");
   printf("getCurrent(B) %d", getCurrent(B));
   printf("\n");
   printf("getIndex(B) %d", getIndex(B));
   deleteCurrent(B);
   printf("\n");
   printListStd(B);
   ListRef D = copyList(B);
   printf("\n");
   printListStd(B);
   printf("\n");
   printListStd(D);
   printf("%s\n", equals(B,D)?"true":"false");



   freeList(&A);
   freeList(&B);
   freeList(&C);
   freeList(&D);

   return(0);
}

File: List Permutations(Java_C)/Project1(Java)/src/ListTest.java
//Eloy Salinas
//esalina1@ucsc.edu | 1243517
//Pa1
//ListTest.java
//Tests the List class functions
public class ListTest {
	public static void main(String[] args){
		List A = new List();
		List B = new List();
		List C = new List();
		List D = new List();
		
		for(int i=1; i<=10; i++){
			A.insertBack(i);
		}
		for(int i=1; i<=10; i++){
			B.insertFront(i);
		}
		for(int i=1; i<=10; i++){
			C.insertFront(i);
		}
		for(int i=1; i<=5; i++){
			D.insertFront(i);
		}
		
		D = A;
		System.out.println("A = " + A);
		System.out.println("B = " + A);
		System.out.println("D = " + D);
		System.out.println("A getFront()" + A.getFront());
		System.out.println("B getFront()" + B.getFront());
		System.out.println("A equals(B)" + A.equals(B) + " Should be false, they were created in different directions.");
		System.out.println("B equals(C)" + B.equals(C) + " Should be true, created the same way.");
		
		A.insertFront(2);
		System.out.println("A = " + A);
		System.out.println(A.getLength());
		
		A.insertBack(5);	
		System.out.println("A = " + A);
		System.out.println(A.getLength());
		
		A.deleteFront();
		System.out.println("A = " + A);
		System.out.println(A.getLength());
		
		A.deleteBack();
		System.out.println("A = " + A);
		System.out.println("A getLength()" + A.getLength());
		System.out.println("A getBack()" + A.getBack());
		System.out.println("A getFront()" + A.getFront());
		System.out.println("A getIndex()" + A.getIndex());
		A.moveTo(9);
		System.out.println("A getCurrent()" + A.getCurrent());
		
		A.moveTo(2);
		System.out.println("A getCurrent()" + A.getCurrent());
		A.moveNext();
		System.out.println("A getCurrent()" + A.getCurrent());
		A.movePrev();
		System.out.println("A getCurrent()" + A.getCurrent());
		
		A.insertAfterCurrent(8);
		System.out.println("A = " + A);
		
		A.insertBeforeCurrent(9);
		System.out.println("A getCurrent()" + A.getCurrent());
		System.out.println("A = " + A);
		
		A.deleteCurrent();
		System.out.println("A = " + A);
		System.out.println("A getLength()" + A.getLength());
		System.out.println("A offEnd()" + A.offEnd());
		
		A.makeEmpty();
		System.out.println("A isEmpty()" + A.isEmpty());
		
	}
	

}
